SQL/MySQL.sublime-syntax

```diff
@@ -622,6 +625,45 @@ contexts:
       scope: keyword.other.dml.sql
       set: maybe-column-name-list
 
+###[ DML LOAD STATEMENTS ]#####################################################
+  dml-load-data:
+    - match: \b(?i:low_priority|concurrent|local)\b
+      scope: storage.modifier.mysql
+    - match: \b(?i:infile)\b
+      scope: keyword.other.dml.mysql
+      set:
+        - dml-load-data-after-infile
+        - meta-filepath
+        - single-string
+
+  dml-load-data-after-infile:
+    - match: \b(?i:ignore|replace)\b
+      scope: storage.modifier.mysql
+    - match: \b(?i:into\s+table)\b
+      scope: keyword.other.dml.mysql
+      push:
+        - maybe-partition
+        - expect-table-name
+    - match: \b(?i:character\s+set)\b
+      scope: storage.modifier.mysql
+      push: charset
+    - match: \b(?i:fields|columns|lines)\b
+      scope: storage.modifier.mysql
+    - match: \b(?i:(optionally\s+)?(?:terminated|enclosed|escaped|starting)\s+by)\b
+      scope: keyword.other.dml.mysql
+      push: single-string
+    - include: numbers
+    - match: (?=;)
+      pop: 1
+
+  meta-filepath:
+    - meta_content_scope: meta.filepath.mysql
+    - include: immediately-pop
+
+  single-string:
+    - include: strings
```

Comment:
Would it make sense to introduce a `string` context right next to `strings` in SQL (Basic) and re-use it here to ensure single string being consumed?


```yml
  string:
    - match: \'
      scope: punctuation.definition.string.begin.sql
      set: inside-single-quoted-string
```

Reply:
I thought about that, and was a little wary of having a duplicated context whose sole change is to perform a `set` rather than a `push`... Plus on a base syntax, where every inheriting syntax also needs to do such repeating contexts when extending one of them, it doesn't seem ideal - it would be easy to forget. I feel like the current solution solves the problem reasonably enough.